### PR TITLE
Added Link component

### PR DESCRIPTION
**Pull request by alexasselin008**

### CODE REVIEW

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
+import { PlusIcon } from "@hopper-ui/icons";
```
Comment (fraincs) [inline packages/components/src/Link/docs/Link.stories.tsx:1]: You could use the Sparkle icon

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    args: {
+        children: [
+            <Text key="1">Learn more</Text>,
+            <PlusIcon key="2" />
+        ]
+    }
+};
+
+/**
+ * *Non standard* start icons can be provided to handle special cases. However, think twice before adding *start* icons, *end* icons should be your go to.
```
Comment (fraincs) [inline packages/components/src/Link/docs/Link.stories.tsx:89]: I think the * is a mistake here: *Non

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+export const Disabled : Story = {
+    ...Variants,
+    args: {
+        ...Variants.args,
+        isDisabled: true
+    }
+};
+
+/**
+ * A link can vary in size.
```
Comment (fraincs) [inline packages/components/src/Link/docs/Link.stories.tsx:129]: Should we expose the inherit option too.

File: packages/components/src/Link/docs/assets/frog.jpg
Comment (fraincs) [inline packages/components/src/Link/docs/assets/frog.jpg:1]: The image could be resized, its almost 500kb

File: packages/components/src/Link/src/Link.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+import { IconListContext } from "../../IconList/index.ts";
+import { Text, TextContext } from "../../Text/index.ts";
+import { SlotProvider, composeClassnameRenderProps, cssModule, isTextOnlyChildren, type SizeAdapter } from "../../utils/index.ts";
+
+import { LinkContext } from "./LinkContext.ts";
+
+import styles from "./Link.module.css";
+
+
```
Comment (fraincs) [inline packages/components/src/Link/src/Link.tsx:15]: two empty line here

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        isExternal: true
+    }
+};
+
+/**
+ * When the colors from the variant props are not enough, you can use the color prop to set the color of the link.
+ * This removes the hover and focus colors from the link.
+ * Use the color props when a link needs to be placed on top of a color background or visual.
+ * Make sure that the background and the link color meet the minimum color contrast ratio.
+ */
```
Comment (fraincs) [inline packages/components/src/Link/docs/Link.stories.tsx:158]: Should we allow to set an hover and focus color?

File: packages/components/src/Link/tests/chromatic/Link.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+                </Div>
+            </Stack>
+        );
+    },
+    args:{
+        href: "#"
+    }
+};
+
+export const StartIcon: Story = {
```
Comment (fraincs) [inline packages/components/src/Link/tests/chromatic/Link.stories.tsx:44]: We should test all size with an icon

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+import { HopperProvider } from "../../HopperProvider/index.ts";
+import { Stack } from "../../layout/index.ts";
+import { Text } from "../../Text/index.ts";
+import { Link } from "../src/Link.tsx";
+
+import { FrogImage } from "./assets/index.ts";
+
+/**
+ * Links allow users to navigate to a different location. They can be presented inline inside a paragraph or as standalone text.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/Link.stories.tsx:14]: Should we mention image links here?

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+    args: {
+        children: [
+            <Text key="1">Learn more</Text>,
+            <SparklesIcon key="2" />
+        ]
+    }
+};
+
+/**
+ * **Non standard** start icons can be provided to handle special cases. However, think twice before adding *start* icons, *end* icons should be your go to.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/Link.stories.tsx:89]: "However, think twice before adding *start* icons"
does the above sound a little threatening? Maybe not, just made me giggle a little.

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+        ...Icon.args,
+        children: [
+            <SparklesIcon key="1" slot="start-icon" />,
+            <Text key="2">Learn more</Text>
+        ]
+    }
+};
+
+/**
+ * A link content can be a single icon.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/Link.stories.tsx:103]:
```suggestion
 * A link's content can be a single icon.
```

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+        ...Icon.args,
+        "aria-label": "Clean",
+        children: [
+            <SparklesIcon key="1" />
+        ]
+    }
+};
+
+/**
+ * A link can have *disabled* style.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/Link.stories.tsx:118]:
```suggestion
 * A link can have a *disabled* style.
```

File: packages/components/src/Link/docs/Link.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+        }
+    ],
+    args: {
+        children: "Go to next router page",
+        href: "/123"
+    }
+};
+
+/**
+ * A link content can be an image element.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/Link.stories.tsx:237]:
```suggestion
 * A link's content can be an image element.
```

File: packages/components/src/Link/docs/migration-notes.md [hunk 12]
```diff
@@ -0,0 +1,2 @@
+Link, IconLink and TextLink has all been merged into one.
```
Comment (vicky-comeau) [inline packages/components/src/Link/docs/migration-notes.md:1]:
```suggestion
Link, IconLink and TextLink have all been merged into one.
```

**Review by vicky-comeau — APPROVED** [review]

**Pull request merged**